torch/distributed/_tensor/debug/visualize_sharding.py

```diff
@@ -0,0 +1,160 @@
+import torch
+from torch.distributed._tensor import DeviceMesh, Shard, Replicate, distribute_tensor
+import os
+
+def shard_or_replicate(curr_ranges, placements, num_splits, idx, grid, device_mesh):
+    if len(placements) == idx:
+        assert not isinstance(device_mesh, list)
+        for i in curr_ranges['row']:
+            for j in curr_ranges['col']:
+                grid[i][j] = f"Device:{str(device_mesh)}" if grid[i][j] == "" else grid[i][j] + f", Device:{str(device_mesh)}"
+        return curr_ranges
+
+    placement = placements[idx]
+
+    next_ranges = []
+    if placement.is_shard():
+        curr_dim = 'row' if placement.dim == 0 else 'col'
+        other_dim = 'row' if placement.dim == 1 else 'col'
+
+        chunk_size, extra = divmod(len(curr_ranges[curr_dim]), num_splits[idx])
```

Comment:
this logic is quite complicated, it looks like we mainly want to get a range of offsets on a specific tensor dimension (either row/col)

can we leverage existing util to compute the ranges to make things simpler? https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/_utils.py#L46